homework01/test/examples/Main.hs

```diff
@@ -1,15 +1,57 @@
-module Main
-  ( main
-  )
-  where
-
--- doctest
-import qualified Test.DocTest as DocTest
+module Main 
+  ( main )
+where
 
+import Basis
+import Test.Hspec
 
 main :: IO ()
-main =
-  DocTest.doctest
-    [ "-isrc"
-    , "src/Basis.hs"
-    ]
+main = hspec $ do
+
+  describe "toDigits" $
+    context "when digits are positive integers" $ do
+      it "converts to a list of digits" $
+        toDigits 1234 `shouldBe` [1, 2, 3, 4]
+
+      it "returns an empty list for 0" $
+        toDigits 0 `shouldBe` []
+
+      it "returns an empty list for negative inputs" $
+        toDigits (-17) `shouldBe` []
+
+  describe "toDigitsRev" $ 
+    context "when digits are positive integers" $ do
+      it "converts to a list of digits reversed" $ 
+        toDigitsRev 9876 `shouldBe` [6, 7, 8, 9]
+
+      it "returns an empty list for 0" $
+        toDigitsRev 0 `shouldBe` []
+
+      it "returns an empty list for negative inputs" $
+        toDigitsRev (-1) `shouldBe` []
+
+  describe "doubleEveryOther" $ do
+    it "doubles every other number beginning from the right (even number of elements)" $
```

Comment:
@jmorocho Perhaps: 

```haskell
context "when the list has an even number of elements"
  it "doubles every other number beginning from the right"  
```

Could you please verify for each case of `doubleEveryOther`?